karapace/schema_reader.py

```diff
@@ -326,8 +325,8 @@ def handle_messages(self) -> None:
                 value = None
                 if msg.value:
                     try:
-                        value = json.loads(msg.value.decode("utf8"))
-                    except json.JSONDecodeError:
+                        value = json_decode(msg.value.decode("utf8"))
```

Comment:
Just one of many places of unnecessary binary decoding, if the signature of `json_decode` is updated as suggested in my other comment.

```suggestion
                        value = json_decode(msg.value)
```

Reply:
Removed the unwanted decoding calls.